### PR TITLE
Egress rules integration tests

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
**What this PR does / why we need it**:
Added tests for egress rules.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```NONE
```

### CODE REVIEW

File: test/integration/testdata/egress-rule-httpbin.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: config.istio.io/v1alpha2
+kind: EgressRule
+metadata:
+  name: httpbin
+spec:
+  destination:
+      service: "httpbin.org"
```
Comment (rshriram) [inline test/integration/testdata/egress-rule-httpbin.yaml.tmpl]: change this to nghttp2.org/httpbin/
and use http2

File: test/integration/testdata/egress-rule-google.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+kind: EgressRule
+metadata:
+  name: google
+spec:
+  destination:
+      service: "*google.com"
+  ports:
+      - port: 443
+        protocol: https
+  use_egress_proxy: false
```
Comment (rshriram) [inline test/integration/testdata/egress-rule-google.yaml.tmpl]: May be enable the egress proxy for one of these?

**Comment by rshriram** [comment]
Thanks for the e2e test!.

File: test/integration/testdata/egress-rule-google.yaml.tmpl (hunk 2 above)
Comment (vadimeisenbergibm) [reply test/integration/testdata/egress-rule-google.yaml.tmpl]: using egress proxy is not implemented - we should implement it in 0.3. It depends on either using nginx or on adding forward proxy functionality to envoy

File: test/integration/testdata/egress-rule-httpbin.yaml.tmpl (hunk 1 above)
Comment (vadimeisenbergibm) [reply test/integration/testdata/egress-rule-httpbin.yaml.tmpl]: hmm, http2 to nghttp2.org seems not to work. Let me debug it tomorrow and add nghttp2.org as an additional case, in a separate PR.

**Pull request merged**